daemon/cmd/daemon_main.go

```diff
@@ -356,6 +356,9 @@ func initializeFlags() {
 	flags.Bool(option.EnableWireguardUserspaceFallback, false, "Enables the fallback to the wireguard userspace implementation")
 	option.BindEnv(Vp, option.EnableWireguardUserspaceFallback)
 
+	flags.String(option.NodeEncryptionOptOutLabels, defaults.NodeEncryptionOptOutLabels, "Label selector for nodes which will opt-out of node-to-node encryption")
+	option.BindEnv(Vp, option.NodeEncryptionOptOutLabels)
```

Comment:
*(Commenting here but actually question on the commit description.)*

> This ensures that all Cilium-managed nodes will be able to reach the kube-apiserver running on that node
regardless of encryption status.

Does that mean that some encryption status can cause a persistent loss of connectivity?

Reply:
The commit and flag descriptions don't explain what opting out of node-to-node encryption means. Does it means excluded_node->node isn't encrypted, or node_excluded->node, or both?

Reply:
Both. If the public key of `excluded_node` is stale, either direction will be broken, so we opt out from both. The way this is implemented is that:

a) `excluded_node` will set it's own EncryptKey in `CiliumNode` to 0, to ensure `node->excluded_node` is not encrypted by the remote node
b)  `excluded_node` will ignore any EncryptKey of remote `CiliumNode`s, to ensure `excluded_node->node` is not encrypted either.

We'll revise the docs to make that a bit more clear.

Reply:
> Does that mean that some encryption status can cause a persistent loss of connectivity?

If a node looses it's private key (which currently is very possible, e.g. happens if the node reboots, since we don't have a persistent storage for private keys at the moment), it will permanently be locked out of the cluster, until a user manually removes it's stale public key from it's CiliumNode CRD.

Reply:
Thanks for the great explanations!